### PR TITLE
[JBPM-9403] Process work-flow breaks for processes with timers if one of Nodes is starting or shutting down at the time of workflow execution

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9403

### CODE REVIEW

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
@Ginxo not idea why this is stack ¿?

**Comment by mareknovotny** [comment]
Jenkins retest this

**Comment by Ginxo** [comment]
sorry @elguardian can you please share the problem link or log? I missed trace

**Comment by mareknovotny** [comment]
Jenḱins do fdb

**Review by afalhambra — COMMENTED** [review]
Very good work, just a minor comment. Also I think it would be great to have a unit test in place for this scenario.

File: jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java [hunk 1]
```diff
@@ -127,15 +127,20 @@ private void executeTimerJobInstance(TimerJobInstance timerJobInstance) throws E
...
     private void recoverTimerJobInstance(EjbTimerJob ejbTimerJob, Exception e) {
-        // if we have next date fired means that it would have been reescheduled already by DefaultTimerJobInstance
         if (ejbTimerJob.getTimerJobInstance().getTrigger().hasNextFireTime() != null) {
-            logger.warn("Execution of time failed Interval Trigger failed {}", ejbTimerJob.getTimerJobInstance());
+            // this is an interval trigger. Problem here is that the timer scheduled by DefaultTimerJobInstance is lost
+            // because of the transaction, so we need to do this here.
+            try {
+                logger.warn("Execution of time failed Interval Trigger failed. Skipping {}", ejbTimerJob.getTimerJobInstance());
+                transaction(job -> internalSchedule(job), ejbTimerJob.getTimerJobInstance());
+            } catch (Exception e1) {
```
Comment (gmunozfe) [inline jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java:141]: perhaps, exception message may also be logged in the warning

**Review by gmunozfe — APPROVED** [review]
Looks good, just a minor comment related to warning logger

File: jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java [hunk 2]
```diff
@@ -127,15 +127,20 @@ private void executeTimerJobInstance(TimerJobInstance timerJobInstance) throws E
...
 
     private void recoverTimerJobInstance(EjbTimerJob ejbTimerJob, Exception e) {
-        // if we have next date fired means that it would have been reescheduled already by DefaultTimerJobInstance
         if (ejbTimerJob.getTimerJobInstance().getTrigger().hasNextFireTime() != null) {
-            logger.warn("Execution of time failed Interval Trigger failed {}", ejbTimerJob.getTimerJobInstance());
+            // this is an interval trigger. Problem here is that the timer scheduled by DefaultTimerJobInstance is lost
+            // because of the transaction, so we need to do this here.
+            try {
+                logger.warn("Execution of time failed Interval Trigger failed. Skipping {}", ejbTimerJob.getTimerJobInstance());
+                transaction(job -> internalSchedule(job), ejbTimerJob.getTimerJobInstance());
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java:140]: done

File: jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java (hunk 1 above)
Comment (elguardian) [reply jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java:141]: done

**Pull request merged**